lib/vagrant-service-manager/services/docker.rb

```diff
@@ -71,6 +69,35 @@ def self.print_env_info(ui, options)
           PluginUtil.print_shell_configure_info(ui, ' docker')
         end
       end
+
+      def self.install_cli(machine, ui, options)
+        Installer.new('docker', machine, ui, options).run
+      end
+
+      def self.binary_path(version)
+        "#{BIN_FOLDER}/docker/#{version}/docker"
+      end
+
+      def self.binary_download_url(machine, version)
+        arch = machine.env.host.capability(:os_arch)
+        "#{BINARY_BASE_URL}/#{os_type}/#{arch}/docker-#{version}"
```

Comment:
I don't think soo. This is just a download url method and I believe each service class should know its download url.
For CDK, this logic will be updated as conditional based generation of url.